Application/Schema.sql

```diff
@@ -41,5 +42,6 @@ CREATE INDEX paragraph_quotes_landing_page_id_index ON paragraph_quotes (landing
 CREATE INDEX paragraph_ctas_landing_page_id_index ON paragraph_ctas (landing_page_id);
 CREATE INDEX paragraph_featured_articles_landing_page_id_index ON paragraph_featured_articles (landing_page_id);
 ALTER TABLE paragraph_ctas ADD CONSTRAINT paragraph_ctas_ref_landing_page_id FOREIGN KEY (landing_page_id) REFERENCES landing_pages (id) ON DELETE NO ACTION;
+ALTER TABLE paragraph_ctas ADD CONSTRAINT paragraph_ctas_ref_ref_landing_page_id FOREIGN KEY (ref_landing_page_id) REFERENCES landing_pages (id) ON DELETE NO ACTION;
```

Comment:
This is causing the error

Reply:
When there's two FKs to the same table, IHP renames the fields to a more verbose form to make it more explicit with FKs you're relating to.

So `landingPage.paragraphCtas` will be now called `landingPage.paragraphCtasLandingPages`. The new FK will be named `landingPage.paragraphCtasRefLandingPages`.

So in your code base you need to rename all references, e.g. here:

```haskell
fetchLandingPageWithParagraphs landingPageId = do
    landingPage :: LandingPage <- fetch landingPageId

    landingPage
        |> pure . modify #paragraphCtasLandingPages (orderByDesc #weight) -- <--- RENAMED HERE
        >>= pure . modify #paragraphQuotes (orderByDesc #weight)
        >>= fetchRelated #paragraphCtasLandingPages
        >>= fetchRelated #paragraphQuotes
````

Reply:
You can use the `Show Haskell Code` function in the Schema Designer to see the `data` code IHP has generated for the table

<img width="554" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/2072185/9715856c-ebce-499f-bc18-903ef300fed4">

This can be useful to debug cases like this :)

Reply:
Thanks :ok: I'll add some docs to the guide about this :smile:

Reply:
Pushed changes, but I'm still doing something wrong, as now I'm getting:

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/85bcd24b-ab46-43a4-832a-868d8340394e)


I tried also keeping the older `Include' ["paragraphCtas", "paragraphQuotes"] LandingPage`, but it didn't work. What is the error message trying to tell me? :thinking:

Reply:
Based on this 

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/ed9a2392-e143-4abd-a7bf-462cfb418a65)

I assume `Include' ["paragraphCtas", "paragraphQuotes"] LandingPage` is the correct form, so pushed it.

The current error message is 

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/39b34c21-ff52-4efe-af4d-3786b0564243)

Reply:
@mpscholten if you get a chance, I'd be happy for a pointer :smile_cat:

Reply:
This error message seems like some type signature in the view is missing the `Include "paragraphCtas"`

Reply:
I think it's just a symptom.

When I remove the code from the Views, then next error is

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/1a691a2e-109d-46c9-9c42-6f731fdcf878)

Reply:
I've removed more views stuff, and that's the error I see

![image](https://github.com/amitaibu/ihp-landing-page/assets/125707/45789919-0c4c-499a-af79-68ce7359573f)

Seems to be expecting `Include "paragraphQuotes" (Include  "paragraphCtasLandingPages"  (FetchResult (Id' "landing_pages") LandingPage))`

Reply:
I've been debugging this and it seems like there's a bug in the generated code by IHP:

<img width="441" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/2072185/9745a97f-3522-4d03-970f-5afd83ac106e">

The two Include's here duplicates

Reply:
Ok! Happy to hear it wasn't just my code, because I was trying hard and couldn't find the solution! :smile:

Reply:
just pushed a failing test and a fix to IHP master 👍  Works now locally

Reply:
Here's the local changes I made:

Added type signatures:

<img width="1224" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/2072185/175a3728-f46a-4c08-8406-ffe6f7e13d0c">


Fixed typo + fixed form (hacky with undefined for now, but atleast it compiles :D ):

<img width="1176" alt="image" src="https://github.com/amitaibu/ihp-landing-page/assets/2072185/abb17e55-f4eb-46b0-8cc3-ee7dd42eba89">